src/System.Net.Sockets/tests/FunctionalTests/LoggingTest.cs

```diff
@@ -34,14 +35,34 @@ public void EventSource_EventsRaisedAsExpected()
                     var events = new ConcurrentQueue<EventWrittenEventArgs>();
                     listener.RunWithCallback(events.Enqueue, () =>
                     {
-                        // Invoke a test that'll cause some events to be generated
+                        // Invoke several tests to execute code paths while tracing is enabled
```

Comment:
> I guess what's kinda unclear to me is why GetAwaiter needs to be called at all.

Do you mean why we need to block at all?  If so, because the calls are asynchronous, and we need to not return from the lambda until all of the work has completed.

Or do you mean why the blocking is done with GetAwaiter().GetResult() rather than with Wait() or Result?  Because Wait() wraps any exception(s) in an AggregateException, and I just want any exception to propagate without being wrapped.   And because these are Task-returning Async calls, so there is no Result (that's on `Task<T>`, since there's no result for the operations.

Or something else?